cs/test/core/ConvertTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace UnitTest.Convert
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading;
+    using Bond;
+    using NUnit.Framework;
+    using Bond.IO.Safe;
+    using Bond.Protocols;
+
+    [global::Bond.Schema]
+    public partial class Decimals
+    {
+        [global::Bond.Id(0), global::Bond.Type(typeof(global::Bond.Tag.blob))]
+        public System.Decimal Price { get; set; }
+
+        [global::Bond.Id(1), global::Bond.Type(typeof(List<global::Bond.Tag.blob>))]
+        public List<decimal> Numbers { get; set; }
+
+        public Decimals()
+            : this("UnitTest.Convert.Decimals", "Decimals")
+        { }
+
+        protected Decimals(string fullName, string name)
+        {
+            Price = new decimal();
+            Numbers = new List<decimal>();
+        }
+    }
+
+    public static class BondTypeAliasConverter
+    {
+        public static int ConvertToDecimalCount = 0;
+        public static int ConvertToArraySegmentCount = 0;
+
+        public static decimal Convert(ArraySegment<byte> value, decimal unused)
+        {
+            var bits = new int[value.Count / sizeof(int)];
+            Buffer.BlockCopy(value.Array, value.Offset, bits, 0, bits.Length * sizeof(int));
+
+            Interlocked.Increment(ref ConvertToDecimalCount);
+
+            return new decimal(bits);
+        }
+
+        public static ArraySegment<byte> Convert(decimal value, ArraySegment<byte> unused)
+        {
+            var bits = decimal.GetBits(value);
+            var data = new byte[bits.Length * sizeof(int)];
+            Buffer.BlockCopy(bits, 0, data, 0, data.Length);
+
+            Interlocked.Increment(ref ConvertToArraySegmentCount);
+
+            return new ArraySegment<byte>(data);
+        }
+
+        public static void Clear()
+        {
+            ConvertToDecimalCount = 0;
+            ConvertToArraySegmentCount = 0;
+        }
+    }
+
+
+    [TestFixture]
+    public class TypeAliasTests
```

Comment:
ConvertTests instead of TypeAliasTests to match file name

Reply:
Done.